bbq/compiler/compiler.go

```diff
@@ -636,41 +644,40 @@ func (c *Compiler[_, _]) exportVariables(variableDecls []*ast.VariableDeclaratio
 }
 
 func (c *Compiler[_, _]) contractType() (contractType sema.CompositeKindedType) {
-	contractDecl := c.Program.SoleContractDeclaration()
-	if contractDecl != nil {
-		contractType = c.DesugaredElaboration.CompositeDeclarationType(contractDecl)
-		return
-	}
-
-	interfaceDecl := c.Program.SoleContractInterfaceDeclaration()
-	if interfaceDecl != nil {
-		contractType = c.DesugaredElaboration.InterfaceDeclarationType(interfaceDecl)
-		return
+	// In tests, there can be contracts along with functions.
+	// TODO: Is it needed to handle having more than one contract?
+	compositeDeclarations := c.Program.CompositeDeclarations()
+	for _, declaration := range compositeDeclarations {
+		if declaration.Kind() == common.CompositeKindContract {
+			return c.DesugaredElaboration.CompositeDeclarationType(declaration)
+		}
```

Comment:
(2) Support compiling locally defined contracts (contracts defined in scripts, needed for tests)